cmd/kubeapps-apis/proto/kubeappsapis/core/packages/v1alpha1/packages.proto

```diff
@@ -306,6 +333,67 @@ message AvailablePackageDetail {
   google.protobuf.Any custom_detail = 13;
 }
 
+// InstalledPackageSummary
+//
+// An InstalledPackageSummary provides a summary of an installed package
+// useful when aggregating many installed packages.
+message InstalledPackageSummary {
+  // TODO: add example for API docs
+  // option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_schema) = {
+  //   example: '{}'
+  // };
+
+  // InstalledPackageReference
+  //
+  // A reference uniquely identifying the package.
+  InstalledPackageReference installed_package_ref = 1;
+
+  // Name
+  //
+  // A name given to the installation of the package (eg. "my-postgresql-for-testing").
+  string name = 2;
+
+  // PkgVersionReference
+  //
+  // The package version reference defines a version or constraint limiting
+  // matching package versions.
+  VersionReference pkg_version_reference = 3;
+
+  // CurrentPkgVersion
+  //
+  // The version of the package which is currently installed.
+  string current_pkg_version = 4;
+
+  // Installed package icon URL
+  //
+  // A url for an icon.
+  string icon_url = 5;
+
+  // PackageDisplayName
+  //
+  // The package name as displayed to users (provided by the package, eg. "PostgreSQL")
+  string pkg_display_name = 6;
+
+  // ShortDescription
+  //
+  // A short description of the package (provided by the package)
+  string short_description = 7;
+
+  // LatestMatchingPkgVersion
+  //
+  // Only non-empty if an available upgrade matches the specified pkg_version_reference.
+  // For example, if the pkg_version_reference is ">10.3.0 < 10.4.0" and 10.3.1
+  // is installed, then:
+  //   * if 10.3.2 is available, latest_matching_version should be 10.3.2, but
+  //   * if 10.4 is available while >10.3.1 is not, this should remain empty.
+  string latest_matching_pkg_version = 8;
+
+  // LatestPkgVersion
+  //
+  // The latest version available for this package, regardless of the pkg_version_reference.
+  string latest_pkg_version = 9;
```

Comment:
Again, flux doesn't seem to provide these 2 in helmrelease.status field, which is where I'd expect to find them. Are these necessary to be able to display whether or not there is an upgrade available? If so, it *SEEMS* to me that the UX or core server can figure these out based on getAvailablePackageVersions. In fact, these two fields seem like they're about "available" more than "installed"

Reply:
> Again, flux doesn't seem to provide these 2 in helmrelease.status field, which is where I'd expect to find them. Are these necessary to be able to display whether or not there is an upgrade available?

Yes, whether an upgrade is available (whether that's in the current version specification, ie. should auto-upgrade unless auto-reconcilliation is paused, or outside the current version specification, in which case a manual upgrade is required if desired.

> If so, it _SEEMS_ to me that the UX or core server can figure these out based on getAvailablePackageVersions. In fact, these two fields seem like they're about "available" more than "installed"

They're about both, at least, the first one is: latest matching pkg version is a function of the available versions that match the specific package version reference used for this installation. Yes, the UX can indeed figure these out based on fetching the available versions for each item in the paginated list and doing the semver work, but it can be quite complex, while (a) it's part of what a client needs to display a list of installed packages to the user, and (b) the data is all available to the plugin. If we did want to remove these from here, I think we'd at least need an API function to get latest matching pkg versions for an input list of (available_pkg_summary_ref, pkg_version_ref) tuples, so only one request is required to get the data for N installed packages that are being displayed.

Thoughts?